stl/inc/numeric

```diff
@@ -70,31 +70,40 @@ inline constexpr bool _Plus_on_arithmetic_ranges_reduction_v = false;
 #endif // _STD_VECTORIZE_WITH_FLOAT_CONTROL
 
 template <class _InIt, class _Ty, class _BinOp>
-_NODISCARD _Ty reduce(const _InIt _First, const _InIt _Last, _Ty _Val, _BinOp _Reduce_op) {
+_NODISCARD _CONSTEXPR20_ICE _Ty reduce(const _InIt _First, const _InIt _Last, _Ty _Val, _BinOp _Reduce_op) {
     // return commutative and associative reduction of _Val and [_First, _Last), using _Reduce_op
     _Adl_verify_range(_First, _Last);
     auto _UFirst      = _Get_unwrapped(_First);
     const auto _ULast = _Get_unwrapped(_Last);
-    if constexpr (_Plus_on_arithmetic_ranges_reduction_v<_Unwrapped_t<const _InIt&>, _Ty, _BinOp>) {
-        (void) _Reduce_op; // TRANSITION, VSO-486357
-        return _Reduce_plus_arithmetic_ranges(_UFirst, _ULast, _Val);
-    } else {
+#ifdef __cpp_lib_is_constant_evaluated
+    if (_STD is_constant_evaluated()) {

```

Comment:
I hate that the effect of `is_constant_evaluated` here is to revert the throughput improvement we get from `if constexpr`. It seems especially unfortunate here given that this function selects between 3 virtually identical loops. I have no concrete suggestions to make, only despair: I can't find a restructuring that reduces the duplication.

Reply:
Err, wait - why do we need this `is_constant_evaluated` branch here at all? Can't we simply make `_Reduce_plus_arithmetic_ranges` `_NODISCARD constexpr`?

Reply:
Derp, I agree. I had _Copy_memmove on the brain.

Reply:
Are you OK if we merge _CONSTEXPR20_ICE in this change anyway since we will need it for the ones in `<xutility>`?

Reply:
Actually we still need this but it gets a TRANSITION comment; C1XX's constexpr evaluator rejects `#pragma loop(ivdep)`.

Reply:
Filed as DevCom-878972

Reply:
> Are you OK if we merge `_CONSTEXPR20_ICE` in this change anyway since we will need it for the ones in `<xutility>`?

Not that it matters now, but for the record, yes I'm fine with that.